pkg/providers/kube/client.go

```diff
@@ -13,10 +13,17 @@ import (
 	"github.com/openservicemesh/osm/pkg/identity"
 	k8s "github.com/openservicemesh/osm/pkg/kubernetes"
 	"github.com/openservicemesh/osm/pkg/service"
+	"github.com/openservicemesh/osm/pkg/utils"
 )
 
-// NewProvider implements mesh.EndpointsProvider, which creates a new Kubernetes cluster/compute provider.
-func NewProvider(kubeController k8s.Controller, providerIdent string, cfg configurator.Configurator) (endpoint.Provider, error) {
+var (
+	// Ensure Client conforms to the provider interfaces
+	_ endpoint.Provider = &Client{}
+	_ service.Provider  = &Client{}
+)
```

Comment:
This seems unnecessary. The compiler should error if the client doesn't implement an interface as long as the functions that require an interface accept the interface type and not the underlying implementation type (Client in this case).